engine/apps/alerts/tasks/distribute_alert.py

```diff
@@ -14,17 +14,23 @@ def distribute_alert(alert_id):
     """
     We need this task to make task processing async and to make sure the task is delivered.
     """
-    from apps.alerts.models import Alert, AlertGroup
+    from apps.alerts.models import Alert
 
     alert = Alert.objects.get(pk=alert_id)
     task_logger.debug(f"Start distribute_alert for alert {alert_id} from alert_group {alert.group_id}")
 
     send_alert_create_signal.apply_async((alert_id,))
-    # If it's the first alert, let's launch the escalation!
+
+    # Launch escalation for the group if it's the first alert, or if the group is paused.
+    # "paused" means that the current escalation step is "Continue escalation if >X alerts per Y minutes" and there are
+    # not enough alerts to trigger the escalation further. Launching escalation for a paused group will re-evaluate
+    # the threshold and advance the escalation if needed, or go back to the same "paused" state if the threshold is
+    # still not reached.
+    if alert.is_the_first_alert_in_group or alert.group.pause_escalation:
+        alert.group.start_escalation_if_needed(countdown=TASK_DELAY_SECONDS)
```

Comment:
The main change of the PR: run `start_escalation_if_needed` not only when receiving the first alert, but also when the group is paused (see comment above).